### PR TITLE
feat: Support sum in window function

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

This PR supports SUM window function for non-decimal data types. For non-decimal types, DataFusion's SUM UDAF will be used. For decimal data type, we probably need to use Comet's own implementation [sum_decimal](https://github.com/apache/datafusion-comet/blob/main/native/core/src/execution/datafusion/expressions/sum_decimal.rs). I will have a separate PR for decimal sum.

### CODE REVIEW

File: native/core/src/execution/datafusion/planner.rs [hunk 1]
```diff
@@ -1277,8 +1277,6 @@ impl PhysicalPlanner {
             }
             AggExprStruct::Min(expr) => {
                 let child = self.create_expr(expr.child.as_ref().unwrap(), schema.clone())?;
-                let datatype = to_arrow_datatype(expr.datatype.as_ref().unwrap());
-                let child = Arc::new(CastExpr::new(child, datatype.clone(), None));
```
Comment (huaxingao) [inline native/core/src/execution/datafusion/planner.rs:1281]: This is not related to window function `sum` support. 
`CastExpr` is not needed for min and max, so remove it.

**Review by kazuyukitanimura — APPROVED** [review]

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 2]
```diff
@@ -210,8 +210,29 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
...
         case agg: AggregateExpression =>
           agg.aggregateFunction match {
-            case _: Min | _: Max | _: Count =>
+            case _: Count =>
               Some(agg)
+            case min: Min =>
+              if (minMaxDataTypeSupported(min.dataType)) {
+                Some(agg)
+              } else {
+                withInfo(windowExpr, "Unsupported datatype", expr)
```
Comment (viirya) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:219]: Can we put the data type in the error message?

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala [hunk 3]
```diff
@@ -1564,14 +1564,72 @@ class CometExecSuite extends CometTestBase {
...
+
+                    val df3 = sql(s"SELECT $function OVER(order by _2 desc) FROM tbl")
+                    checkSparkAnswerWithTol(df3, 1e-6)
+
+                    val df4 = sql(s"SELECT $function OVER(partition by _2 order by _2) FROM tbl")
+                    checkSparkAnswerWithTol(df4, 1e-6)
+                  }
+                }
+
+                // SUM doesn't work for Date type. org.apache.spark.sql.AnalysisException will be thrown.
```
Comment (viirya) [inline spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:1596]: Is it because DataFusion doesn't support it?

Comment (viirya) [reply spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:1596]: If so, can we open an issue and put the issue link here?

File: native/core/src/execution/datafusion/planner.rs [hunk 4]
```diff
@@ -1640,32 +1640,39 @@ impl PhysicalPlanner {
...
 
-    fn process_agg_func(agg_func: &AggExpr) -> Result<(String, Vec<Expr>), ExecutionError> {
-        fn optional_expr_to_vec(expr_option: &Option<Expr>) -> Vec<Expr> {
-            expr_option
-                .as_ref()
-                .cloned()
-                .map_or_else(Vec::new, |e| vec![e])
-        }
-
-        fn int_to_stats_type(value: i32) -> Option<StatsType> {
```
Comment (viirya) [inline native/core/src/execution/datafusion/planner.rs:1651]: We don't need this?

Comment (viirya) [reply native/core/src/execution/datafusion/planner.rs:1651]: Is it just not used?

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala (hunk 2 above)
Comment (huaxingao) [reply spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala]: Fixed. Thanks

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala (hunk 3 above)
Comment (huaxingao) [reply spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:1596]: Actually Spark throws Exception for Sum with Date type.
```
org.apache.spark.sql.AnalysisException: [DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE] Cannot resolve "sum(_12)" due to data type mismatch: Parameter 1 requires the "NUMERIC" or "ANSI INTERVAL" type, however "_12" has the type "DATE".; line 1 pos 7;
'Project [unresolvedalias(sum(_12#11) windowspecdefinition(unspecifiedframe$()), None)]
```

File: native/core/src/execution/datafusion/planner.rs (hunk 4 above)
Comment (huaxingao) [reply native/core/src/execution/datafusion/planner.rs:1651]: Yea, it's not used so I remove this

**Review by viirya — APPROVED** [review]

**Comment by huaxingao** [comment]
Merged. Thanks @kazuyukitanimura @viirya
